connectors/byoei.py

```diff
@@ -177,7 +177,7 @@ def __init__(
         queue,
         index,
         existing_ids,
-        filtering=Filtering(),
+        filtering=Filter(),
```

Comment:
if you use `Filter` instead of `Filtering`, shouldn't we rename the `filtering` argument as well?

Reply:
I know it's a bit confusing, we actually use `filtering` for everything interchangeably. 

This is the structure of the connector filtering:
```
"filtering": [
  {
    "domain": "DEFAULT",
    "draft": {"advanced_snippet": {}, "rules": [], "validation": {}},
    "active": {"advanced_snippet": {}, "rules": [], "validation": {}}
  }
]
```

The `Filtering` instance represents the outer array, but there should only be one element, which we filter by `domain`, and the `Filter` instance represents either the `draft` or `active` filtering. 

I think we should have some proper naming. cc @timgrein `Filterings` and `Filtering`? I don't think it's a good idea to name a class plural.

Reply:
Maybe we should go for `FilterCollection`, which contains multiple `Filter` instances. The assumption that we always only have one `Filter` instance inside `filtering` will probably not be true anymore in 8.8 as we may look into using separate `Filter` instances for different tables (as an example).

++ on class names shouldn't be plural, would make it harder to distinguish between `Filters` and `Filter`

https://medium.com/swlh/an-opinionated-guide-to-naming-your-code-aimed-at-new-developers-6f769d60e098 mentions some further reasons under "Class names"

Reply:
I kind of think should rename `Filter` to `Filtering`, as this is the first and only place I see it's named like this.

In the Project brief, its official name is `Filtering Rules`. In UI, it's Sync Rules.

WDYT if we rename `Filtering` to `FilteringSetting`, and rename `Filter` to `Filtering`.

Speaking of plural class name, we do have class `PipelineSettings`.

Reply:
My personal preference would be to have a class name, which makes it obvious that we're having multiple instances of a specific type at hand. A `FilteringSetting` could also be _one_ filtering setting for _one_ domain. From the workplace search docs I see that they're using the term _filters_. Maybe @danajuratoni can help, how we'll call a single filter/filtering/filtering setting (or whatever) in the future and base our decision on that?

But for me it's fine as long as it's consistent and obvious what's the container and what's the single instance. I don't want to block here 👍

Reply:
Agree, I don't think this is blocking. Not customer-facing. I will just rename this as `Filter`.